index.js

```diff
@@ -0,0 +1,380 @@
+					res.render("requestUpdate", {
+						res: req.session.user,
+						bookingData,
+						eventsData
+					});
+				}
+			);
+		});
+	} else {
+		res.redirect("/login");
+	}
+});
+app.get("/logout", (req, res) => {
+	res.clearCookie("session", {
+		path: "/"
+	});
+	res.redirect("/login");
+});
+app.post("/onLogin", (req, res) => {
+	var username = req.body.username;
+	var password = req.body.password;
+	con.query(
+		"select * from users where username='" + username + "';",
+		(err, rows) => {
+			if (!err) {
+				if (rows.length > 0) {
+					if (
+						password ===
+						AES.decrypt(rows[0].password, `${process.env.AES_key}`)
+					) {
+						req.session.user = rows[0];
+						res.redirect("/dashboard");
+					} else {
+						console.log("Invalid password");
+						res.status(401).render("error", {
+							errorMessage: "Authentication error!"
+						});
+					}
+				} else {
+					console.log("Username not found!");
+					res.status(401).render("error", {
+						errorMessage: "Authentication error!"
+					});
+				}
+			} else {
+				console.log("Username not found!");
+				res.status(401).render("error", {
+					errorMessage: "Authentication error!"
+				});
+			}
+		}
+	);
+});
+app.post("/checkAvailability", (req, res) => {
+	if (req.session.user) {
+		con.query(
+			"SELECT slotHall FROM booking WHERE event_id IN (SELECT id FROM events WHERE date_wanted = ?)",
+			req.body.date_wanted,
+			(err, result) => {
+				if (!err) {
+					res.send(result);
+				} else {
+					console.log(err);
+					res.status(500).render("error", {
+						errorMessage: "Availability checking failed!"
+					});
+				}
+			}
+		);
+	} else {
+		res.redirect("/login");
+	}
+});
+app.post("/makeRequest", (req, res) => {
+	if (req.session.user) {
+		var userID = req.session.user.id;
+		var date_wanted = req.body.date_wanted;
+		var slotHall = "";
+		for (var i = 1; i <= 7; i++)
+			for (var j = 1; j <= 9; j++) {
+				if (req.body[`cell${j}${i}`]) {
+					slotHall += j.toString() + i.toString();
+				}
+			}
+		var club = req.body.club_name;
+		var details = req.body.desc;
+		var title = req.body.event_name;
+		var requirements = req.body.requirements;
+		var event_obj = {
+			title: title,
+			club: club,
+			requirements: requirements,
+			details: details,
+			date_wanted: date_wanted,
+			userID: userID
+		};
+		con.query("insert into events set ?", event_obj, err => {
+			if (!err) {
+				console.log(`Inserted ${event_obj.title} into events.`);
+				con.query(
+					"select id from events WHERE userID= ? ORDER BY id DESC LIMIT 1",
+					req.session.user.id,
+					(err, event_id) => {
+						if (!err) {
+							var booking_obj = {
+								userID: userID,
+								event_id: event_id[0].id,
+								slotHall: slotHall
+							};
+							con.query(
+								"insert into booking set ?",
+								booking_obj,
+								err => {
+									if (!err) {
+										console.log(
+											`Inserted event with id ${booking_obj.event_id} into booking.`
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'booking_obj' is not in camel case. (camelcase)](https://app.codacy.com/app/AravindVNair99/Hall-Management-System/pullRequest?prid=4592231)